src/java_bytecode/generate_java_generic_type.cpp

```diff
@@ -70,8 +80,9 @@ symbolt generate_java_generic_typet::operator()(
     pre_modification_size==after_modification_size,
     "All components in the original class should be in the new class");
 
-  const java_class_typet &new_java_class = construct_specialised_generic_type(
-    generic_class_definition, new_tag, replacement_components);
+  const java_specialised_generic_class_typet &new_java_class =
```

Comment:
Get by value (the return is by-value, and the reference relies on flaky lifetime-extension magic)